apps/dashboard/src/components/workflow-editor/steps/conditions/edit-step-conditions-form.tsx

```diff
@@ -0,0 +1,155 @@
+import { Form, FormField } from '@/components/primitives/form/form';
+import { parseStepVariables } from '@/utils/parseStepVariablesToLiquidVariables';
+import { updateStepInWorkflow } from '@/components/workflow-editor/step-utils';
+import { useWorkflow } from '@/components/workflow-editor/workflow-provider';
+import { UnsavedChangesAlertDialog } from '@/components/unsaved-changes-alert-dialog';
+import { useBeforeUnload } from '@/hooks/use-before-unload';
+
+const rulesSchema: z.ZodType<RuleType | RuleGroupType> = z.union([
+  z
+    .object({
+      field: z.string().min(1),
+      operator: z.string(),
+      value: z.string().nullable(),
+    })
+    .passthrough()
+    .superRefine(({ operator, value }, ctx) => {
+      if (operator === 'between' || operator === 'notBetween') {
+        const values = value?.split(',').filter((val) => val.trim() !== '');
+        if (!values || values.length !== 2) {
+          ctx.addIssue({ code: z.ZodIssueCode.custom, message: 'Both values are required', path: ['value'] });
+        }
+      } else if (operator !== 'null' && operator !== 'notNull') {
+        const trimmedValue = value?.trim();
+        if (!trimmedValue || trimmedValue.length === 0) {
+          ctx.addIssue({ code: z.ZodIssueCode.custom, message: 'Value is required', path: ['value'] });
+        }
+      }
+    }),
+  z
+    .object({
+      combinator: z.string(),
+      rules: z.array(z.lazy(() => rulesSchema)),
+    })
+    .passthrough(),
+]);
+
+type FormQuery = {
+  query: RuleGroupType;
+};
+
+const conditionsSchema: z.ZodType<FormQuery> = z.object({
+  query: z
+    .object({
+      combinator: z.string(),
+      rules: z.array(rulesSchema),
+    })
+    .passthrough(),
+});
+
+export const EditStepConditionsForm = () => {
+  const { workflow, step, update } = useWorkflow();
+  const query = useMemo(
+    () =>
+      // prepareRuleGroup and parseJsonLogic calls are needed to generate the unique ids on the query and rules,
+      // otherwise the lib will do it and it will result in the form being dirty
+      step?.controls.values.skip
+        ? prepareRuleGroup(parseJsonLogic(step.controls.values.skip as RQBJsonLogic))
+        : prepareRuleGroup({ combinator: 'and', rules: [] }),
+    [step]
+  );
+  const form = useForm<FormQuery>({
+    mode: 'onSubmit',
+    resolver: zodResolver(conditionsSchema),
+    defaultValues: {
+      query,
+    },
+  });
+  const { formState } = form;
+  const blocker = useBlocker(formState.isDirty);
+  useBeforeUnload(formState.isDirty);
+
+  const { fields, variables } = useMemo(() => {
+    if (!step) return { fields: [], variables: [] };
+
+    const parsedVariables = parseStepVariables(step.variables);
+    return {
+      fields: parsedVariables.primitives.map((primitive) => ({
+        name: primitive.label,
+        label: primitive.label,
+        value: primitive.label,
+      })),
+      variables: [...parsedVariables.primitives, ...parsedVariables.namespaces],
+    };
```

Comment:
parse `fields` and `variables` the values used for the condition `field` and `value` inputs accordingly...
`fields` are just primitives, but `values` are primitives + namespaces